libvips/convolution/im_aconv.c

```diff
@@ -1005,9 +1005,7 @@ aconv_horizontal( Boxes *boxes, IMAGE *in, IMAGE *out )
 
 #define CLIP_UCHAR( V ) \
 G_STMT_START { \
-	if( (V) < 0 ) \
-		(V) = 0; \
-	else if( (V) > UCHAR_MAX ) \
+	if( (V) > UCHAR_MAX ) \
```

Comment:
I don't think this is right, it needs to clip <0 ints to 0 when casting down to uchar.

Perhaps your warning is triggered by L1146? That's using uchar for a char case, I'm not sure why. Perhaps changing that line instead would be safer?